api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/AddressTest.java

```diff
@@ -0,0 +1,112 @@
+package uk.gov.dhsc.htbhf.claimant.entity;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.nio.CharBuffer;
+import java.util.Set;
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+
+import static uk.gov.dhsc.htbhf.claimant.assertion.ConstraintViolationAssert.assertThat;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.AddressTestDataFactory.aValidAddress;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.AddressTestDataFactory.anAddressWithAddressLine1;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.AddressTestDataFactory.anAddressWithAddressLine2;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.AddressTestDataFactory.anAddressWithPostcode;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.AddressTestDataFactory.anAddressWithTownOrCity;
+
+public class AddressTest {
+
+    // Create a string 501 characters long
+    private static final String LONG_STRING = CharBuffer.allocate(501).toString().replace('\0', 'A');
+
+    private Validator validator;
+
+    @BeforeEach
+    void setup() {
```

Comment:
Might it be worthwhile having a `AbstractValidationTest` and move this into there?